src/artemis_driver.cpp

```diff
@@ -234,10 +234,15 @@ TaskCollection ArtemisDriver<GEOM>::StepTasks() {
                                 time, bdt);
       }
 
+      // Apply pgen-defined source term
+      // NOTE(@pdmullen): Intentional positioning after S(W) srcs and before S(U) srcs
```

Comment:
`S(W)`: source terms computed from (begin-stage) primitives:
`S(U)`: source terms computed from (partially-updated) conserved variables. 

for example in the 1st stage of Heun's method

U1* = U0  + dt divF[U0] + dt S[W]
U1 = U1* + dt S[U1*]

Reply:
We could clarify this and call it `ExplicitProblemSourceTerm`, but not a big deal.

Reply:
The current positioning does permit you to make this the "first" S(U) source if you really like.

Reply:
Chatted on the phone -- I was momentarily confused about meshdata registers vs prim/cons state vectors.

Also I think this can still be `ImplicitProblemSourceTerm` since we can't for example stop people from doing a backward Euler update inside this call, so I think keeping the name as-is is appropriate for now.